src/Generators/AbstractClassGenerator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Blueprint\Generators;
+
+use Blueprint\Blueprint;
+use Blueprint\Concerns\HandlesImports;
+use Blueprint\Concerns\HandlesTraits;
+use Blueprint\Contracts\Model;
+use Blueprint\Tree;
+use Illuminate\Filesystem\Filesystem;
+
+class AbstractClassGenerator
+{
+    use HandlesImports, HandlesTraits;
+
+    const INDENT = '        ';
+
+    protected Filesystem $filesystem;
+
+    private Tree $tree;
+
+    protected array $imports = [];
```

Comment:
Is it possible for these respective property to be added to the `trait` and not the abstract class?